src/scenes/AcceleratorRouter/Scoring/EditView.tsx

```diff
@@ -0,0 +1,126 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { Box, Grid, useTheme } from '@mui/material';
+import { makeStyles } from '@mui/styles';
+import { Button } from '@terraware/web-components';
+
+import Page from 'src/components/Page';
+import Card from 'src/components/common/Card';
+import PageForm from 'src/components/common/PageForm';
+import { APP_PATHS } from 'src/constants';
+import strings from 'src/strings';
+import { Score, ScoreValue } from 'src/types/Score';
+
+import PhaseScores from './PhaseScores';
+import { useScoringData } from './ScoringContext';
+import useScoresUpdate from './useScoresUpdate';
+
+const useStyles = makeStyles(() => ({
+  form: {
+    width: '100%',
+  },
+}));
+
+const ScorecardEditView = () => {
+  const theme = useTheme();
+  const history = useHistory();
+  const classes = useStyles();
+  const { crumbs, phase0Scorecard, phase1Scorecard, projectId, projectName, status } = useScoringData();
+  const { update, status: updateStatus } = useScoresUpdate(projectId);
+
+  const [scores, setScores] = useState<Score[]>([]);
+
+  const goToVoting = useCallback(() => {
+    history.push({ pathname: APP_PATHS.ACCELERATOR_VOTING.replace(':projectId', `${projectId}`) });
+  }, [history, projectId]);
+
+  const goToScorecardView = useCallback(() => {
+    history.push({ pathname: APP_PATHS.ACCELERATOR_SCORING.replace(':projectId', `${projectId}`) });
+  }, [history, projectId]);
+
+  const onCancel = useCallback(() => {
+    goToScorecardView();
+  }, [goToScorecardView]);
+
+  const handleOnChange = (key: 'value' | 'qualitative', score: Score, value: ScoreValue | string) => {
+    setScores((prev) =>
+      prev.map((_score: Score) => {
+        if (_score.category !== score.category) {
+          return _score;
+        }
+        return {
+          ..._score,
+          [key]: value,
+        };
+      })
+    );
+  };
+
+  const handleOnChangeValue = (score: Score, value: ScoreValue) => handleOnChange('value', score, value);
+
+  const handleOnChangeQualitative = (score: Score, value: string) => {
+    handleOnChange('qualitative', score, value);
+  };
+
+  const onSave = useCallback(() => {
+    // For now we can only save Phase 1 scores
+    update('Phase 1 - Feasibility Study', scores);
```

Comment:
The backend API only expects diffs for update requests. This looks the same if only one person is using the API at a time but there can be scenarios where two people have the edit page at the same time, and end up overwriting the other one's changes. 

I am wondering if we can keep a separate `updatedScores` state. Same applies to voting I think.

Reply:
Good call, I'll put scores state into the view so we only push updated scores.